nautobot/extras/plugins/validators.py

```diff
@@ -2,40 +2,31 @@
 
 from django.apps import apps
 
-from nautobot.data_validation.custom_validators import BaseValidator
 from nautobot.extras.plugins import CustomValidator
 from nautobot.extras.registry import registry
 from nautobot.extras.utils import FeatureQuery
 
 
-def custom_validator_clean(model_clean_func):
+def custom_validator_clean(model_class):
     """
     Decorator that wraps a models existing clean method to also execute registered plugin custom validators
 
-    :param model_clean_func: The original model clean method which is to be wrapped
+    Args:
+        model_class: The model class whose clean function is to be wrapped
     """
 
+    model_clean_func = model_class.clean
+    model_name = model_class._meta.label_lower
+
     @wraps(model_clean_func)
     def wrapper(model_instance):
         # Run original model clean method
         model_clean_func(model_instance)
 
-        # Run registered plugin custom validators
-        model_name = model_instance._meta.label_lower
-
-        # Note this registry holds instances of PluginCustomValidator registered from plugins
+        # Run App registered custom validators
+        # Note this registry holds instances of CustomValidator registered from plugins
         # which is different than the `custom_validators` model features registry
-        custom_validators = registry["plugin_custom_validators"].get(model_name, [])
-
-        # Add the BaseValidator from data validation engine to the list of custom validators
-        dve_base_validator = type(
-            f"{model_name.split('.')[0].capitalize()}{model_name.split('.')[1].capitalize()}CustomValidator",
-            (BaseValidator,),
-            {"model": model_name},
-        )
-        custom_validators.append(dve_base_validator)
```

Comment:
This was the "main" bug after all - note that we were getting `custom_validators` from the registry, and calling `append()` on it. Two problems:

- this meant that *every time the wrapped `clean()` was called*, we **added another BaseValidator class to the registry list** (oof!)
- BaseValidator subclasses were already being created and registered in `DataValidationEngineConfig.ready()` so I discovered that this was completely redundant even after fixing the erroneous `append()` call. Hence I've removed it entirely.